### PR TITLE
github pr support changes

### PR DESCRIPTION
**Pull request by birmacher**
### New Pull Request Checklist

- [x] Run `go fmt` on your files (e.g. `go fmt ./service/common.go`, or on the whole `service` folder: `go fmt ./service/...`)
- [x] Write tests for your code
  - The tests should cover both "success" and "error" cases.
  - The tests should also check **all the returned variables**, don't ignore any returned value!
  - Ideally the tests should be easily readable, we usually use tests to document our code instead of code comments.
    An example, if you'd write a comment like "Given X this function will return Y" or
    "Beware, if the input is X this function will return Y" then you should implement this as
    a unit test, instead of writing it as a comment.
- [x] If your Pull Request is more than a bug fix you should also check `README.md` and change/add the descriptions there - also
  feel free to add yourself as a contributor if you implement support for a new service ;)
- [x] Before creating the Pull Request you should also run `bitrise run test` with the [Bitrise CLI](https://www.bitrise.io/cli),
  to perform all the automatic checks (which will run on your Pull Request when you open it).

### Summary of Pull Request

GH Pull Request changes to expose source, destination branch with the merge branch

### CODE REVIEW

File: bitriseapi/bitriseapi.go [hunk 1]
```diff
@@ -21,13 +21,16 @@ type EnvironmentItem struct {
...
-	CommitMessage string            `json:"commit_message,omitempty"`
-	Branch        string            `json:"branch,omitempty"`
-	Tag           string            `json:"tag,omitempty"`
-	PullRequestID *int              `json:"pull_request_id,omitempty"`
-	WorkflowID    string            `json:"workflow_id,omitempty"`
-	Environments  []EnvironmentItem `json:"environments,omitempty"`
+	CommitHash               string            `json:"commit_hash,omitempty"`
+	CommitMessage            string            `json:"commit_message,omitempty"`
+	Branch                   string            `json:"branch,omitempty"`
+	BranchDest               string            `json:"branch_dest,omitempty"`
```
Comment (viktorbenei) [inline bitriseapi/bitriseapi.go]: comment

File: bitriseapi/bitriseapi.go [hunk 2]
```diff
@@ -21,13 +21,16 @@ type EnvironmentItem struct {
...
+	Tag                      string            `json:"tag,omitempty"`
+	PullRequestID            *int              `json:"pull_request_id,omitempty"`
+	PullRequestRepositoryURL string            `json:"pull_request_repository_url,omitempty"`
+	PullRequestMergeBranch   string            `json:"pull_request_merge_branch,omitempty"`
```
Comment (viktorbenei) [inline bitriseapi/bitriseapi.go]: comment

File: service/hook/github/github.go [hunk 3]
```diff
@@ -13,6 +13,13 @@ import (
 	"github.com/bitrise-io/go-utils/sliceutil"
 )
 
+const (
+	// ContentTypeApplicationJSON ...
+	ContentTypeApplicationJSON string = "application/json"
```
Comment (viktorbenei) [inline service/hook/github/github.go]: could be moved into a common package

File: service/hook/github/github.go [hunk 4]
```diff
@@ -34,15 +41,19 @@ type CodePushEventModel struct {
...
 
 // PullRequestInfoModel ...
 type PullRequestInfoModel struct {
-	BranchInfo BranchInfoModel `json:"head"`
-	Title      string          `json:"title"`
-	Body       string          `json:"body"`
-	Merged     bool            `json:"merged"`
-	Mergeable  *bool           `json:"mergeable"`
+	HeadBranchInfo BranchInfoModel `json:"head"`
+	BaseBranchInfo BranchInfoModel `json:"base"`
```
Comment (viktorbenei) [inline service/hook/github/github.go]: comment - what's the difference?

File: service/hook/github/github.go [hunk 5]
```diff
@@ -130,10 +141,13 @@ func transformPullRequestEvent(pullRequest PullRequestEventModel) hookCommon.Tra
...
-					CommitMessage: commitMsg,
-					Branch:        pullRequest.PullRequestInfo.BranchInfo.Ref,
-					PullRequestID: &pullRequest.PullRequestID,
+					CommitMessage:            commitMsg,
+					CommitHash:               pullRequest.PullRequestInfo.HeadBranchInfo.CommitHash,
+					Branch:                   pullRequest.PullRequestInfo.HeadBranchInfo.Ref,
+					BranchDest:               pullRequest.PullRequestInfo.BaseBranchInfo.Ref,
+					PullRequestID:            &pullRequest.PullRequestID,
+					PullRequestRepositoryURL: pullRequest.PullRequestInfo.HeadBranchInfo.getRepositoryURL(),
+					PullRequestMergeBranch:   fmt.Sprintf("pull/%v/merge", pullRequest.PullRequestID),
```
Comment (viktorbenei) [inline service/hook/github/github.go]: why `%v`? shouldn't this be `%d`?

File: service/hook/github/github.go [hunk 6]
```diff
@@ -236,3 +250,10 @@ func (hp HookProvider) TransformRequest(r *http.Request) hookCommon.TransformRes
 		Error: fmt.Errorf("Unsupported GitHub event type: %s", ghEvent),
 	}
 }
+
+func (branchInfoModel BranchInfoModel) getRepositoryURL() string {
```
Comment (viktorbenei) [inline service/hook/github/github.go]: please add a comment why this works like this

File: service/hook/github/github.go (hunk 4 above)
Comment (birmacher) [inline service/hook/github/github.go]: one of them is base while the other one is head ;)

Comment (viktorbenei) [inline service/hook/github/github.go]: yeah but which one's which ;)

Comment (birmacher) [inline service/hook/github/github.go]: that's a secret ;) - updated

File: bitriseapi/bitriseapi.go [hunk 7]
```diff
@@ -21,16 +21,16 @@ type EnvironmentItem struct {
...
-	CommitMessage            string            `json:"commit_message,omitempty"`
-	Branch                   string            `json:"branch,omitempty"`
-	BranchDest               string            `json:"branch_dest,omitempty"`
-	Tag                      string            `json:"tag,omitempty"`
-	PullRequestID            *int              `json:"pull_request_id,omitempty"`
-	PullRequestRepositoryURL string            `json:"pull_request_repository_url,omitempty"`
-	PullRequestMergeBranch   string            `json:"pull_request_merge_branch,omitempty"`
-	WorkflowID               string            `json:"workflow_id,omitempty"`
-	Environments             []EnvironmentItem `json:"environments,omitempty"`
+	CommitHash               string            `json:"commit_hash,omitempty"`                 // git commit hash
```
Comment (viktorbenei) [inline bitriseapi/bitriseapi.go]: please comment _above_ the line

File: service/hook/github/github.go [hunk 8]
```diff
@@ -48,8 +41,8 @@ type BranchInfoModel struct {
 
 // PullRequestInfoModel ...
 type PullRequestInfoModel struct {
-	HeadBranchInfo BranchInfoModel `json:"head"`
-	BaseBranchInfo BranchInfoModel `json:"base"`
+	HeadBranchInfo BranchInfoModel `json:"head"` // source brach for the pull request
```
Comment (viktorbenei) [inline service/hook/github/github.go]: comment above the line

File: service/hook/github/github.go [hunk 9]
```diff
@@ -251,6 +244,7 @@ func (hp HookProvider) TransformRequest(r *http.Request) hookCommon.TransformRes
 	}
 }
 
+// returns the repository clone URL depending on the publicity of the project
```
Comment (viktorbenei) [inline service/hook/github/github.go]: what's `SSHURL` and what's `CloneURL`? is `CloneURL` always the public / HTTPS url and SSH the `git@..` ssh/private URL?

File: service/hook/github/github.go [hunk 10]
```diff
@@ -34,15 +34,19 @@ type CodePushEventModel struct {
 type BranchInfoModel struct {
 	Ref        string `json:"ref"`
 	CommitHash string `json:"sha"`
+	Private    bool   `json:"private"`
+	SSHURL     string `json:"ssh_url"`
```
Comment (viktorbenei) [inline service/hook/github/github.go]: what's the difference between `CloneURL` and `SSHURL`?

**Pull request merged**